File: 2399. Check Distances Between Same Letters/solution.py
class Solution(object):
  def checkDistances(self, s, distance):
    """
    :type s: str
    :type distance: List[int]
    :rtype: bool
    """
    for index, letter in enumerate(s):
      letterIndex = ord(letter) - 97
      distanceValue = distance[letterIndex]
      nextValue = index + distanceValue + 1
      if len(s) <= nextValue:
        return False
      if letter == s[nextValue]:
        distance[letterIndex] = -1
      elif distance[letterIndex] != -1:
        return False
    return True

File: 2399. Check Distances Between Same Letters/solution.cpp
class Solution {
public:
  bool checkDistances(string s, vector<int>& distance) {
    for (int index = 0; index < s.size(); index++){
      char letter = s[index];
      int letterIndex = (int) letter - 97;
      int distanceValue = distance[letterIndex];
      int nextValue = index + distanceValue + 1;
      if (s.size() <= nextValue){
        return false;
      }
      if (letter == s[nextValue]){
        distance[letterIndex] = -1;
      }
      else if (distance[letterIndex] != -1){
        return false;
      }
    }
    return true;
  }
};